### PR TITLE
1412 add a storybook framework for nextjs to our pr deployments

### PR DESCRIPTION
**Pull request by zxh722**
## Summary of the changes
Created a nextjs storybook that will improve our ability to test server side rendering, this storybook has also been added to comments within a PR.

## Related issue
#1412 

## Checklist
- [x] Relevant unit tests and visual regression tests added. 
- [x] Visual testing against Figma component specification completed. 
- [x] All acceptance criteria reviewed and met. 
- [x] Accessibility Insights FastPass performed.
- [x] A11y unit test added and yields no issues.
- [x] A11y plug-in on Storybook yields no issues. 
- [x] Manual screen reader testing performed using NVDA and VoiceOver. 
- [x] Page can be zoomed to 400% with no loss of content. 
- [x] Screen magnifier used with no issues. 
- [x] Text resized to 200% with no loss of content.
- [x] Text spacing increased as per the [WCAG 1.4.12 success criterion](https://www.w3.org/TR/WCAG21/#text-spacing) with no loss of content.
- [x] Browser setting 'prefers reduced motion' tested. No animations or motion visible whilst this setting is on. 
- [x] Windows High Contrast mode tested with no loss of content.
- [x] System light and dark mode tested with no loss of content. 
- [x] Manual keyboard testing for keyboard controls and logical focus order. 
- [x] Min/max content examples tested with no loss of content or overflow. 
- [x] Browser support tested (Chrome, Safari, Firefox and Edge).
- [x] Correct roles used and ARIA attributes used correctly where required. 
- [x] Logical heading structure is maintained, and the HTML elements used for headings can be changed to fit within the wider page structure. 
- [x] All prop combinations work without issue. 
- [x] Changes to docs package checked and committed.
- [x] Tested for FOUC (Flash of Unstyled Content) in both SSR (Server-Side Rendering) and SSG (Static Site Generation) settings.

### CODE REVIEW

**Review by GCHQ-Developer-112 — CHANGES_REQUESTED** [review]
Looks good! 

A few things I've noticed:
- There are loading errors with the docs for the following components: breadcrumbs & hero
- There are no props showing for select
- The default theme isn't being applied to our themed components i.e. side nav, top nav etc. The default seems to be the sunset theme
- Changing the theme from the `ic-theme` component page doesn't affect the theme of themed components

**Comment by zxh722** [comment]
> Looks good!
> 
> A few things I've noticed:
> 
> * There are loading errors with the docs for the following components: breadcrumbs & hero
> * There are no props showing for select
> * The default theme isn't being applied to our themed components i.e. side nav, top nav etc. The default seems to be the sunset theme
> * Changing the theme from the `ic-theme` component page doesn't affect the theme of themed components

First two bullet points have been resolved, @GCHQ-Developer-112 is creating a new issue for the remaining two bullet points.

File: packages/nextjs/README.md [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+## Learn More
+
+To learn more about Next.js, take a look at the following resources:
+
+- [Next.js Documentation](https://nextjs.org/docs) - learn about Next.js features and API.
+- [Learn Next.js](https://nextjs.org/learn) - an interactive Next.js tutorial.
+
+You can check out [the Next.js GitHub repository](https://github.com/vercel/next.js/) - your feedback and contributions are welcome!
+
+## Deploy on Vercel
```
Comment (GCHQ-Developer-299) [inline packages/nextjs/README.md:32]: Probably should strip out the default NextJS readme stuff here, like mention of Vercel

**Review by GCHQ-Developer-299 — CHANGES_REQUESTED** [review]

**Review by GCHQ-Developer-299 — APPROVED** [review]

**Comment by GCHQ-Developer-112** [comment]
The changes look good! Final comment, please can you move the change to the package.json that's currently in the nextjs commit to the root commit

**Review by GCHQ-Developer-112 — DISMISSED** [review]

**Review by GCHQ-Developer-299 — APPROVED** [review]

**Review by GCHQ-Developer-112 — APPROVED** [review]

**Pull request merged**